zookeeper/source/zk_nn_client.cc

```diff
@@ -721,6 +854,44 @@ void ZkNnClient::complete(CompleteRequestProto& req,
       res.set_result(false);
       return;
   }
+
+  // Remove this client from the leases branch.
+  bool exists;
+  std::string client_name = req.clientname();
+  if (client_name.size() != 0) {
+    if (!zk->exists(LeaseZookeeperPath(src) + '/' +
+                      client_name, exists, error_code)) {
+      LOG(ERROR) << "Failed to check the existence of "
+      << LeaseZookeeperPath(src) + '/' + client_name << ".";
+      res.set_result(false);
+      return;
+    }
+    if (exists) {
+      if (!zk->delete_node(LeaseZookeeperPath(src) + '/'
```

Comment:
Is this repeated code from the RecoverLease method? Wouldn't it make sense to modularize out a "delete lease" method? (To be used by both the RecoverLease method and this method here)